pkg/helm/chartproxy/proxy.go

```diff
@@ -69,13 +69,15 @@ func (p *proxy) IndexFile() (*repo.IndexFile, error) {
 		return nil, err
 	}
 
-	var indexFiles []*repo.IndexFile
+	indexFile := repo.NewIndexFile()
 	for _, helmRepo := range helmRepos {
 		idxFile, err := helmRepo.IndexFile()
 		if err != nil {
 			return nil, err
 		}
-		indexFiles = append(indexFiles, idxFile)
+		for key, entry := range idxFile.Entries {
+			indexFile.Entries[key+"--"+helmRepo.Name] = entry
```

Comment:
Maybe we add '~~' as a delimiter between the two names to make it more unique?
```suggestion
			indexFile.Entries[key+"~~"+helmRepo.Name] = entry
```

Reply:
this char is not alowed - only letters, digits and dash

Reply:
Okay, no problem.